### PR TITLE
Add DNS provider for Bunny

### LINKED ISSUE #1759: Support for provider: bunny.net
### Welcome

- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, the DNS provider exposes a public API.
- [X] Yes, I know that the lego maintainers don't have an account in all DNS providers in the world.
- [ ] Yes, I'm able to create a pull request and be able to maintain the implementation.
- [X] Yes, I'm able to test an implementation if someone creates a pull request to add the support of this DNS provider.

### How do you use lego?

Through Traefik

### Link to the DNS provider

https://bunny.net/

### Link to the API documentation

https://docs.bunny.net/reference/dnszonepublic_index

### Additional Notes

Bunny launched DNS and I find it very cool. 
There is already a project to do this manual :
[https://github.com/mwt/certbot-dns-bunny](url)

But it would be very cool to have this automated with traefik and other proxies.
Maybe I can test the DNS Provider with some notes how I can compile it.

### PR DESCRIPTION
**Pull request by TECHNOFAB11**
adds support for the Bunny DNS provider. Uses the Bunny Go client from https://github.com/simplesurance/bunny-go.
just took the Hetzner provider and switched out the important parts :) 

Fixes #1759

### CODE REVIEW

**Comment by ldez** [comment]
Hello, in order for a PR adding a DNS provider to be accepted, you have to:

- [ ] add a description to your PR
- [ ] be able to maintain this provider
- [ ] have a homogeneous design with the other providers
- [ ] add tests (units)
```shell
make test
```
- [ ] add tests ("live")
https://github.com/go-acme/lego/blob/c938de68f262c057542d44bcb67ef961cefcc60e/providers/dns/cloudflare/cloudflare_test.go#L125-L151

```shell
make test
```
- [ ] add a [provider descriptor](https://github.com/go-acme/lego/blob/master/providers/dns/cloudflare/cloudflare.toml)
- [ ] generate CLI help, documentation, and readme.
```bash
make generate-dns
```
- [ ] be able to do: _(and put the output of this command to a comment in your PR)_
```bash
rm -rf .lego

./lego -m your@email.com --dns YOUR_PROVIDER_NAME -d *.example.com -d example.com -s https://acme-staging-v02.api.letsencrypt.org/directory run
```
Note the wildcard domain is important.
- [ ] pass the linter ([golangci-lint](https://github.com/golangci/golangci-lint#install) must be installed):
```shell
make checks
```
- [ ] do `go mod tidy`

File: providers/dns/bunny/bunny.toml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+Name = "Bunny"
+Description = ''''''
+URL = "https://bunny.net"
+Code = "bunny"
+Since = "v4.10.0"
```
Comment (ldez) [inline providers/dns/bunny/bunny.toml:5]:
```suggestion
Since = "v4.11.0"
```

File: go.mod [hunk 2]
```diff
@@ -71,6 +71,8 @@ require (
 	software.sslmate.com/src/go-pkcs12 v0.2.0
 )
 
+require github.com/simplesurance/bunny-go v0.0.0-20221115111006-e11d9dc91f04
```
Comment (ldez) [inline go.mod:74]: can you move this element into the previous `require` block?

**Review by ldez — CHANGES_REQUESTED** [review]

**Comment by TECHNOFAB11** [comment]
help & docs commit will follow shortly. Output of the command, example.com was replaced with example.<domain>.de and parts are masked:

```
lego --email XXXXX --dns bunny -d *.example.XXXXX.de -d example.XXXXX.de -s https://acme-staging-v02.api.letsencrypt.org/directory run
```

```
2023/02/22 20:12:54 Please review the TOS at https://letsencrypt.org/documents/LE-SA-v1.3-September-21-2022.pdf
Do you accept the TOS? Y/n
Y
2023/02/22 20:12:56 [INFO] acme: Registering account for XXXXXXX
!!!! HEADS UP !!!!

Your account credentials have been saved in your Let's Encrypt
configuration directory at "/XXXXXXXXXX/lego/.lego/accounts".

You should make a secure backup of this folder now. This
configuration directory will also contain certificates and
private keys obtained from Let's Encrypt so making regular
backups of this folder is ideal.
2023/02/22 20:12:56 [INFO] [*.example.XXXXX.de, example.XXXXX.de] acme: Obtaining bundled SAN certificate
2023/02/22 20:12:57 [INFO] [*.example.XXXXX.de] AuthURL: https://acme-staging-v02.api.letsencrypt.org/acme/authz-v3/5477863934
2023/02/22 20:12:57 [INFO] [example.XXXXX.de] AuthURL: https://acme-staging-v02.api.letsencrypt.org/acme/authz-v3/5477863944
2023/02/22 20:12:57 [INFO] [*.example.XXXXX.de] acme: use dns-01 solver
2023/02/22 20:12:57 [INFO] [example.XXXXX.de] acme: Could not find solver for: tls-alpn-01
2023/02/22 20:12:57 [INFO] [example.XXXXX.de] acme: Could not find solver for: http-01
2023/02/22 20:12:57 [INFO] [example.XXXXX.de] acme: use dns-01 solver
2023/02/22 20:12:57 [INFO] [*.example.XXXXX.de] acme: Preparing to solve DNS-01
2023/02/22 20:12:58 [INFO] [example.XXXXX.de] acme: Preparing to solve DNS-01
2023/02/22 20:12:58 [INFO] [*.example.XXXXX.de] acme: Trying to solve DNS-01
2023/02/22 20:12:58 [INFO] [*.example.XXXXX.de] acme: Checking DNS record propagation using [XXXXXX:53 [XXXX::XXXX:XXXX:XXXX:XX%eth0]:53]
2023/02/22 20:13:00 [INFO] Wait for propagation [timeout: 2m0s, interval: 2s]
2023/02/22 20:13:08 [INFO] [*.example.XXXXX.de] The server validated our request
2023/02/22 20:13:08 [INFO] [example.XXXXX.de] acme: Trying to solve DNS-01
2023/02/22 20:13:08 [INFO] [example.XXXXX.de] acme: Checking DNS record propagation using [XXXXXX:53 [XXXX::XXXX:XXXX:XXXX:XX%eth0]:53]
2023/02/22 20:13:10 [INFO] Wait for propagation [timeout: 2m0s, interval: 2s]
2023/02/22 20:13:11 [INFO] [example.XXXXX.de] The server validated our request
2023/02/22 20:13:11 [INFO] [*.example.XXXXX.de] acme: Cleaning DNS-01 challenge
2023/02/22 20:13:12 [INFO] [example.XXXXX.de] acme: Cleaning DNS-01 challenge
2023/02/22 20:13:12 [INFO] [*.example.XXXXX.de, example.XXXXX.de] acme: Validations succeeded; requesting certificates
2023/02/22 20:13:12 [INFO] [*.example.XXXXX.de] Server responded with a certificate.
```

Will clean things up in a seperate commit

File: go.mod (hunk 2 above)
Comment (TECHNOFAB11) [reply go.mod:74]: Sure, didn't see GoLand put that in its own block. Weird

**Comment by ldez** [comment]
to answer the following checkbox:
> [ ] be able to maintain this provider

I need to know more about you: are you an employee of BunnyWay or a customer?

**Comment by TECHNOFAB11** [comment]
I'm a customer & most maintenance is already handled by the external library so I can maintain this if needed. Not very active on Github though tbh, might need  a mention if something's wrong

**Comment by ldez** [comment]
Can you generate the doc again? You made modifications after the generation :wink:

**Review by ldez — APPROVED** [review]
LGTM

**Pull request merged**